src/sparkx/Particle.py

```diff
@@ -246,15 +249,21 @@ def __init__(
             (input_format is None) and (particle_array is not None)
         ):
             raise ValueError("'input_format' or 'particle_array' not given")
+        
+        if attribute_list is None and input_format == "ASCIICustom":
+            raise ValueError("'attribute_list' not given")
+
+        if attribute_list!=[] and input_format != "ASCIICustom":
+            raise ValueError("'OscarCustom' format requires 'attribute_list' to be given.")
 
         if (input_format is not None) and (particle_array is not None):
-            self.__initialize_from_array(input_format, particle_array)
+            self.__initialize_from_array(input_format, particle_array, attribute_list)
 
     def __initialize_from_array(
-        self, input_format: str, particle_array: np.ndarray
+        self, input_format: str, particle_array: np.ndarray, attribute_list: List[str]
     ) -> None:
         """
-        Initialize instance attributes based on the provided input format and array.
+        Initialize instance attributes based on the provided input format and array, and optionally an atttribute list.
 
         Parameters
         ----------
```

Comment:
The new `input_format` string is missing in the function documentation.

Reply:
Fixed.